Dockerfile

```diff
@@ -0,0 +1,48 @@
+# Use Node.js LTS (matches engine requirement of >=18)
+FROM node:23-slim
+
+# Install system dependencies
+RUN apt-get update && apt-get install -y \
+    python3 \
+    make \
+    g++ \
+    bash \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install pnpm (matches packageManager version)
+RUN npm install -g pnpm@9.0.0
+
+# Set working directory
+WORKDIR /app
+
+# Copy package files
+COPY package.json pnpm-workspace.yaml turbo.json ./
+COPY clients/client-coinbase/package.json ./clients/client-coinbase/
+COPY clients/client-direct/package.json ./clients/client-direct/
+COPY clients/client-github/package.json ./clients/client-github/
+COPY plugins/plugin-0x/package.json ./plugins/plugin-0x/
+COPY plugins/plugin-coinbase/package.json ./plugins/plugin-coinbase/
+COPY plugins/plugin-coinbase/advanced-sdk-ts/package.json ./plugins/plugin-coinbase/advanced-sdk-ts/
+COPY plugins/plugin-email/package.json ./plugins/plugin-email/
+COPY plugins/plugin-github/package.json ./plugins/plugin-github/
+COPY plugins/plugin-twitter/package.json ./plugins/plugin-twitter/
+COPY agent/package.json ./agent/
+COPY ui/package.json ./ui/
```

Comment:
I see that we do not copy other source files (other than mostly package.json files) into the image. Why is that?